ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java

```diff
@@ -183,6 +183,24 @@ public void init() {
         }
     }
 
+    private void mapComponentsToHostGroups(String defaultBindTo, List<String> componentNames) {
+        for (String componentName : componentNames) {
+            if(componentName.contains("|")) {
+                String[] split = componentName.split("\\|");
+                if (!componentsByNode.containsKey(split[1])) {
+                    componentsByNode.put(split[1], new MutableList<String>());
+                }
+                componentsByNode.get(split[1]).add(split[0]);
```

Comment:
As a sanity check, shouldn't we perform a `trim()` against `split[0]` and `split[1]`?